firestore/integration_test_internal/src/android/arena_ref_android_test.cc

```diff
@@ -42,89 +59,114 @@ TEST_F(ArenaRefTestAndroid, ConstructFromEnvAndObject) {
-  EXPECT_TRUE(arena_ref4.get(env).Equals(env, string1));
+  EXPECT_TRUE(arena_ref2.get(env).Equals(env, string1));
 
-  arena_ref3 = arena_ref1;
-  EXPECT_EQ(arena_ref3.get(env).get(), nullptr);
-  EXPECT_TRUE(arena_ref4.get(env).Equals(env, string1));
+  arena_ref2 = arena_ref1;
+  EXPECT_EQ(arena_ref2.get(env).get(), nullptr);
+  EXPECT_TRUE(arena_ref3.get(env).Equals(env, string1));
 }
 
-TEST_F(ArenaRefTestAndroid, MoveConstructor) {
+TEST_F(ArenaRefTestAndroid, MovesReferenceToNull) {
   Env env;
 
   ArenaRef arena_ref1;
   ArenaRef arena_ref2(std::move(arena_ref1));
   EXPECT_EQ(arena_ref1.get(env).get(), nullptr);
   EXPECT_EQ(arena_ref2.get(env).get(), nullptr);
+}
 
-  Local<String> string = env.NewStringUtf("hello world");
+TEST_F(ArenaRefTestAndroid, MovesReferenceToValidObject) {
+  Env env;
 
-  ArenaRef arena_ref3(env, string);
-  ArenaRef arena_ref4(std::move(arena_ref3));
-  EXPECT_EQ(arena_ref3.get(env).get(), nullptr);
-  EXPECT_TRUE(arena_ref4.get(env).Equals(env, string));
+  Local<String> string = env.NewStringUtf("hello world");
 
-  ArenaRef arena_ref5 = std::move(arena_ref4);
-  EXPECT_TRUE(arena_ref5.get(env).Equals(env, string));
+  ArenaRef arena_ref2(env, string);
+  ArenaRef arena_ref3(std::move(arena_ref2));
+  EXPECT_EQ(arena_ref2.get(env).get(), nullptr);
+  EXPECT_TRUE(arena_ref3.get(env).Equals(env, string));
 }
 
-TEST_F(ArenaRefTestAndroid, MoveAssignmentOp) {
+TEST_F(ArenaRefTestAndroid, MoveAssignsReferenceToNull) {
   Env env;
 
   ArenaRef arena_ref1, arena_ref2;
   arena_ref2 = std::move(arena_ref1);
   EXPECT_EQ(arena_ref1.get(env).get(), nullptr);
   EXPECT_EQ(arena_ref2.get(env).get(), nullptr);
+}
+
+TEST_F(ArenaRefTestAndroid, MoveAssignsReferenceToValidObject) {
+  Env env;
 
   Local<String> string1 = env.NewStringUtf("hello world");
   Local<String> string2 = env.NewStringUtf("hello earth");
 
-  ArenaRef arena_ref3(env, string1);
-  arena_ref3 = std::move(arena_ref3);
+  ArenaRef arena_ref1;
+  ArenaRef arena_ref2(env, string1);
+  arena_ref2 = std::move(arena_ref2);
+  EXPECT_TRUE(arena_ref2.get(env).Equals(env, string1));
+
+  ArenaRef arena_ref3(env, string2);
+  arena_ref3 = std::move(arena_ref2);
+  EXPECT_EQ(arena_ref2.get(env).get(), nullptr);
   EXPECT_TRUE(arena_ref3.get(env).Equals(env, string1));
 
-  ArenaRef arena_ref4(env, string2);
-  arena_ref4 = std::move(arena_ref3);
+  arena_ref3 = std::move(arena_ref1);
   EXPECT_EQ(arena_ref3.get(env).get(), nullptr);
-  EXPECT_TRUE(arena_ref4.get(env).Equals(env, string1));
+}
+
+TEST_F(ArenaRefTestAndroid, EnvCallTakeArenaRefTest) {
+  Env env;
 
-  arena_ref4 = std::move(arena_ref1);
-  EXPECT_EQ(arena_ref4.get(env).get(), nullptr);
+  ArenaRef hash_map(env, HashMap::Create(env));
+  Local<Long> key = Long::Create(env, 1);
+  Local<Long> val = Long::Create(env, 2);
+  env.Call(hash_map, kPut, key, val);
+  Local<Object> result = env.Call(hash_map, kGet, key);
+  EXPECT_TRUE(result.Equals(env, val));
+  EXPECT_TRUE(env.ok());
```

Comment:
This line, `EXPECT_TRUE(env.ok()` can be removed.

The FirestoreAndroidIntegrationTest will fail the test if the test ends with env.ok()==false.

https://github.com/firebase/firebase-cpp-sdk/blob/e61d6bb264633c720b1ce857717f4e9638f40279/firestore/integration_test_internal/src/android/firestore_integration_test_android.cc#L70-L80